### PR TITLE
fix: transaction controller patch

### PR DESCRIPTION
**Pull request by blackdevelopa**
## **Description**
The [fix: legacy gas fee edit modal PR](https://github.com/MetaMask/metamask-mobile/pull/7035#discussion_r1334646147) introduced changes to the [monorepo](https://github.com/MetaMask/core/pull/1672) and wasn't patched properly. This PR fixes that. [Context](https://consensys.slack.com/archives/C02U025CVU4/p1695634752325359?thread_ts=1695403712.510279&cid=C02U025CVU4)

_Fixes #???_
https://github.com/MetaMask/metamask-mobile/pull/7035#discussion_r1334646147

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: patches/@metamask+transaction-controller+4.0.1.patch [hunk 1]
```diff
@@ -2843,7 +2857,7 @@ index dc4300f..65d6703 100644
      maxFeePerGas: (maxFeePerGas) => (0, ethereumjs_util_1.addHexPrefix)(maxFeePerGas),
      maxPriorityFeePerGas: (maxPriorityFeePerGas) => (0, ethereumjs_util_1.addHexPrefix)(maxPriorityFeePerGas),
      estimatedBaseFee: (maxPriorityFeePerGas) => (0, ethereumjs_util_1.addHexPrefix)(maxPriorityFeePerGas),
-+    type: (type) => type === '0x0' ? '0x0' : undefined,
++    type: (type) => (type === '0x0' ? '0x0' : undefined),
  };
  /**
   * Return a URL that can be used to fetch ETH transactions.
```
Comment (jpuri) [inline patches/@metamask+transaction-controller+4.0.1.patch:2863]: Hey @blackdevelopa : can you plz double check to ensure that this file is generated by running build from your core repo branch to create the patch.

Comment (blackdevelopa) [reply patches/@metamask+transaction-controller+4.0.1.patch:2863]: Yes. I think this is a lint fix from the core.

**Review by jpuri — APPROVED** [review]
LGTM

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by Gudahtt** [comment]
Passing E2E smoke test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/1f7b1a43-b929-48d8-8f54-6ed5154588ec

**Pull request merged**